http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -289,6 +295,42 @@ class DataprocInterpreter[F[_]: Parallel](
             None
           )
         )
+
+        // If the cluster is configured with worker private access, then remove the private access
+        // network tag from the master node as soon as possible.
+        //
+        // We intentionally do this here instead of DataprocRuntimeMonitor to more quickly remove
+        // the tag because the init script may depend on public Internet access.
+        _ <- if (machineConfig.workerPrivateAccess) {
+          val op = for {
+            dataprocInstances <- googleDataprocService
+              .getClusterInstances(
+                params.runtimeProjectAndName.googleProject,
+                machineConfig.region,
+                DataprocClusterName(params.runtimeProjectAndName.runtimeName.asString)
+              )
+            masterComputeInstance <- dataprocInstances.find(_._1.role == DataprocRole.Master).flatTraverse {
+              case (DataprocRoleZonePreemptibility(_, z, _), is) =>
+                is.headOption.flatTraverse { i =>
+                  googleComputeService.getInstance(params.runtimeProjectAndName.googleProject, z, i)
+                }
+            }
+            op <- masterComputeInstance.traverse { m =>
```

Comment:
I wonder if we can just assmue master instance name to be `<clusterName>-m` 🤔

Reply:
Maybe.. although we need to call `getInstance` anyway to get the fingerprint to pass to `setInstanceTags`

Reply:
oh I see...